### ISSUE TITLE
[New Service]: AWS Clean Rooms

### ISSUE THREAD

**Issue by rmalecky** [hashicorp/terraform-provider-aws#30024]
### Description

AWS Clean Rooms is an analytics service that allows organizations to query their combined datasets without granting unrestricted access to their raw data. 

Users create [collaborations](https://docs.aws.amazon.com/clean-rooms/latest/apireference/API_CreateCollaboration.html) that include one or more AWS accounts.  Each AWS account in the collaboration creates a [membership](https://docs.aws.amazon.com/clean-rooms/latest/apireference/API_CreateMembership.html) to link the account to a collaboration.  

In parallel users create [configured tables](https://docs.aws.amazon.com/clean-rooms/latest/apireference/API_CreateConfiguredTable.html) that reference datasets stored in Amazon S3  and cataloged in the [AWS Glue data catalog](https://docs.aws.amazon.com/glue/latest/dg/catalog-and-crawler.html).  The user applies an [analysis rule](https://docs.aws.amazon.com/clean-rooms/latest/apireference/API_CreateConfiguredTableAnalysisRule.html) to each configured table to control how the dataset can be used in queries.  

The last step in setting up AWS Clean Rooms is to create [associations between a configured tables and a collaborations](https://docs.aws.amazon.com/clean-rooms/latest/apireference/API_CreateConfiguredTableAssociation.html).

Once the resources are setup collaboration members can use the StartProtectedQuery API to run SQL queries against the APIs.  Runner queries is not in scope for Terraform.

### Requested Resource(s) and/or Data Source(s)

`aws_cleanrooms_collaboration`
`aws_cleanrooms_configured_table`
`aws_cleanrooms_configured_table_analysis_rule`
`aws_cleanrooms_configured_table_association`
`aws_cleanrooms_membership`

### References

[AWS Clean Rooms user guide](https://docs.aws.amazon.com/clean-rooms/latest/userguide/what-is.html)
[AWS Clean Rooms API reference](https://docs.aws.amazon.com/clean-rooms/latest/apireference/Welcome.html)

### Would you like to implement a fix?
yes
